src/lib/gadgets/rot.ts

```diff
@@ -0,0 +1,118 @@
+import { Field } from '../field.js';
+import { Provable } from '../provable.js';
+import { Fp } from '../../bindings/crypto/finite_field.js';
+import * as Gates from '../gates.js';
+
+export { rot, rotate };
+
+const MAX_BITS = 64 as const;
+
+function rot(field: Field, bits: number, direction: 'left' | 'right' = 'left') {
+  // Check that the rotation bits are in range
+  if (bits < 0 || bits > MAX_BITS) {
+    throw Error(`rot: expected bits to be between 0 and 64, got ${bits}`);
```

Comment:
Actually, I would have said `<= 0 and => MAX_BITS`, given that rotation in those cases is trivial, and those could be excluded from the circuit wlog. The current implementation in Kimchi does not support =0 nor =64, for efficiency. Thus, I would say we should either adapt the Kimchi rotation witness generation code to support those edge cases (just some types adaptation to prevent overflows, the actual constraints remain unchanged), or else this interface should behave just like the other one. In any case, I just created a [Kimchi PR](https://github.com/o1-labs/proof-systems/pull/1299) to support this functionality, so both implementations do the same.

Reply:
Ah, nice! Thank you for making the Kimchi PR. I don't mind changing the checks; I guess I'll just wait until the Kimchi PR lands before merging this.

Reply:
The PR has just landed. Feel free to include the support for =0 and =64. Nonetheless, it would be a good idea to point out in a comment or the docs, that rotation by those amounts is trivial and they should reason about whether they really want to include those rows in the circuit or not.